node/src/main/scala/coop/rchain/node/web/package.scala

```diff
@@ -1,71 +1,81 @@
 package coop.rchain.node
 
-import cats.effect.{ConcurrentEffect, ExitCode, Timer}
+import cats.effect.{ConcurrentEffect, ExitCode, Sync, Timer}
 import cats.syntax.all._
 import coop.rchain.comm.discovery.NodeDiscovery
 import coop.rchain.comm.rp.Connect.{ConnectionsCell, RPConfAsk}
 import coop.rchain.node.api.{AdminWebApi, WebApi}
 import coop.rchain.node.diagnostics.NewPrometheusReporter
 import coop.rchain.node.effects.EventConsumer
 import coop.rchain.node.web.ReportingRoutes.ReportingHttpRoutes
+import coop.rchain.node.web.https4s.RouterFix
 import coop.rchain.shared.Log
 import monix.execution.Scheduler
 import org.http4s.HttpRoutes
 import org.http4s.implicits._
-import org.http4s.server.Router
 import org.http4s.server.blaze.BlazeServerBuilder
 import org.http4s.server.middleware.CORS
 
 import scala.concurrent.duration.{DurationInt, FiniteDuration}
 
 package object web {
+  // https://github.com/http4s/http4s/security/advisories/GHSA-52cf-226f-rhr6
+  //  val corsPolicy = CORS.policy.withAllowCredentials(false) // after http4s v0.22.x
+  def corsPolicy[F[_]: Sync](routes: HttpRoutes[F]) =
+    CORS(routes, CORS.DefaultCORSConfig.copy(allowCredentials = false))
+
   def aquireHttpServer[F[_]: ConcurrentEffect: Timer: RPConfAsk: NodeDiscovery: ConnectionsCell: EventConsumer: Log](
       reporting: Boolean,
       host: String = "0.0.0.0",
       httpPort: Int,
       prometheusReporter: NewPrometheusReporter,
       connectionIdleTimeout: FiniteDuration,
-      webApiRoutes: WebApi[F],
+      webApi: WebApi[F],
       reportingRoutes: ReportingHttpRoutes[F]
   )(implicit scheduler: Scheduler): F[fs2.Stream[F, ExitCode]] =
     for {
       event              <- EventsInfo.service[F]
       reportingRoutesOpt = if (reporting) reportingRoutes else HttpRoutes.empty
       baseRoutes = Map(
-        "/metrics"   -> CORS(NewPrometheusReporter.service[F](prometheusReporter)),
-        "/version"   -> CORS(VersionInfo.service[F]),
-        "/status"    -> CORS(StatusInfo.service[F]),
-        "/ws/events" -> CORS(event),
-        "/api"       -> CORS(WebApiRoutes.service[F](webApiRoutes) <+> reportingRoutesOpt)
+        "/metrics"   -> corsPolicy(NewPrometheusReporter.service[F](prometheusReporter)),
+        "/version"   -> corsPolicy(VersionInfo.service[F]),
+        "/status"    -> corsPolicy(StatusInfo.service[F]),
+        "/ws/events" -> corsPolicy(event),
+        "/api"       -> corsPolicy(WebApiRoutes.service[F](webApi) <+> reportingRoutesOpt),
+        // Web API v1 with OpenAPI schema
+        "/api/v1" -> corsPolicy(WebApiRoutesV1.create[F](webApi))
```

Comment:
I'm using prefix `v1` for this version instead of v2 because this is the first web API implementation which is done properly with schema generation.

@zsluedem, @dckc Does this make sense to you?

Reply:
No. I am not aware of any requirement to use schema generation for an OpenAPI. And there are certainly clients that use v1.

Reply:
@dckc Sorry, I was not clear enough about my question.

Existing API which is accessible on `/api` endpoint will not be changed, but may not be updated with the new features/methods.

The new API will have all the same methods at the start, but with auto generated OpenAPI schema accessible via API also.
So there will be no breaking change on existing API (except what will be necessary for block-merge, but this is not the subject of this PR).

The new version of API, because it's basically the same, it's exposed as `/api/v1` and not *v2* because later we will unify API schema for gRPC and HTTP and this will be **version 2** of API.

Does this make sense?

Reply:
Do you want to put `v2` in `/api` in the future?